pkg/rancher-desktop/main/update/MSIUpdater.ts

```diff
@@ -0,0 +1,154 @@
+        'ERR_UPDATER_INVALID_UPDATE_INFO');
+    }
+
+    return this.executeDownload({
+      fileExtension: 'msi',
+      fileInfo,
+      downloadUpdateOptions,
+      task:          async(destinationFile, downloadOptions, packageFile, removeTempDirIfAny) => {
+        const httpExecutor: ElectronHttpExecutor = (this as any).httpExecutor;
+        const mergedDownloadOptions = {
+          ...downloadOptions,
+          sha512: fileInfo.packageInfo?.sha512 ?? downloadOptions.sha512,
+        };
+
+        this._logger.debug?.(`Downloading update for ${ info.version } from ${ fileInfo.url } (sha512: ${ mergedDownloadOptions.sha512 })`);
+        await httpExecutor.download(fileInfo.url, destinationFile, mergedDownloadOptions);
+        const signatureError = await this.verifySignature_(destinationFile);
+
+        if (signatureError) {
+          await removeTempDirIfAny();
+          throw newError(
+            `New version ${ info.version } (${ fileInfo.url }) is not signed correctly: ${ signatureError }`,
+            'ERR_UPDATER_INVALID_SIGNATURE');
+        }
+      },
+    });
+  }
+
+  // Verify that the given file is signed by the expected entity (as configured
+  // in electron-builder.yml).
+  private async verifySignature_(destinationFile: string): Promise<string | null> {
+    let publisherName: string | Array<string> | null;
+
+    try {
+      const configOnDisk: Lazy<any> = (this as any).configOnDisk;
+
+      publisherName = (await configOnDisk.value).publisherName;
+      if (!publisherName) {
+        return null;
+      }
+    } catch (e: any) {
+      if (e?.code === 'ENOENT') {
+        return null; // No updates configured
+      }
+      throw e;
+    }
+    const publisherNames = Array.isArray(publisherName) ? publisherName : [publisherName];
+
+    return await verifySignature(publisherNames, destinationFile, this._logger);
+  }
+
+  protected doInstall(options: InstallOptions): boolean {
+    const systemRoot = process.env.SystemRoot ?? 'C:\\Windows';
+    const msiexec = path.join(systemRoot, 'system32', 'msiexec.exe');
+    const args: string[] = [
+      '/norestart',
+      '/lv*', path.join(paths.logs, 'msiexec.log'),
+      '/i', options.installerPath,
+    ];
+
+    if (options.isSilent) {
+      args.push('/quiet');
+    } else {
+      args.push('/passive');
+    }
+
+    args.push(`MSIINSTALLPERUSER=${ options.isAdminRightsRequired || this.shouldElevate ? '0' : '1' }`);
+
+    if (options.isForceRunAfter) {
+      args.push('RDRUNAFTERINSTALL=1');
+    }
+
+    this._logger.debug?.(`Will invoke installer on restart with: msiexec ${ args.join(' ') }`);
+    mainEvents.on('quit', () => {
+      this._logger.debug?.(`Running msiexec ${ args.join(' ') }`);
+      const proc = spawn(msiexec, args, {
+        detached: true, stdio: 'ignore', windowsHide: true,
+      });
+
+      proc.on('error', (err: NodeJS.ErrnoException) => {
+        this._logger.info(`Cannot run installer: error code: ${ err.code }`);
```

Comment:
Is this meant to be an error or info?

Reply:
Oops, `error` would be more correct; fixed in b8397db37c3b8e845e0779fa7796e495c00df7fc.